### PR TITLE
adding image needed for aws tests

### PR DESCRIPTION
**Pull request by RinkiyaKeDad**
Signed-off-by: Arsh Sharma <arshsharma461@gmail.com>

Based on @irbekrm's suggestion here https://github.com/jetstack/testing/pull/512#discussion_r679861495, this PR adds the image we need to run the AWS tests.

### CODE REVIEW

File: images/golang-aws/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    ./aws/install; \
+    curl -o aws-iam-authenticator https://amazon-eks.s3.us-west-2.amazonaws.com/1.19.6/2021-01-05/bin/linux/amd64/aws-iam-authenticator; \
+    chmod +x ./aws-iam-authenticator; \
+    mv aws-iam-authenticator /usr/local/bin/; \
+    curl -O https://releases.hashicorp.com/terraform/1.0.3/terraform_1.0.3_linux_amd64.zip; \
+    unzip terraform_1.0.3_linux_amd64.zip;  \
+    mv terraform /usr/local/bin/; \
+    go get github.com/bazelbuild/bazelisk; \
+    bazelisk; \
+    ln -s /go/bin/bazelisk /usr/local/bin/bazel;
```
Comment (irbekrm) [inline images/golang-aws/Dockerfile:16]: Nit: new line

File: images/golang-aws/build.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+name: golang-aws # Name of the image to be built
+
+# Image names to be tagged and pushed
+images:
+- ${_REGISTRY}/${_NAME}:${_DATE_STAMP}-${_GIT_REF}
+- ${_REGISTRY}/${_NAME}:latest
```
Comment (irbekrm) [inline images/golang-aws/build.yaml:6]: Nit: new line

File: images/golang-aws/build.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+name: golang-aws # Name of the image to be built
+
```
Comment (irbekrm) [inline images/golang-aws/build.yaml:2]: Suggestion: add a variants section and tag the images with the version of Go this is based on like it's done for golang-dind image https://github.com/jetstack/testing/blob/master/images/golang-dind/build.yaml#L8 and parameterize the Go version in the Dockerfile too.

**Review by irbekrm — CHANGES_REQUESTED** [review]
Looks good to me! I've added two nits and a suggestion to parameterize the Go version and tag the image with that.

**Review by jakexks — DISMISSED** [review]
/lgtm

File: images/golang-aws/build.yaml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  "1.16.6":
+    arguments:
+      BASE_IMAGE: "golang:1.16.6-buster"
+      TERRAFORM_VERSION: "1.0.3"
+      BAZEL_VERSION: "4.0.0"
+
+# Image names to be tagged and pushed
+images:
+- ${_REGISTRY}/${_NAME}:${_DATE_STAMP}-${_GIT_REF}-${GO_VERSION}
+- ${_REGISTRY}/${_NAME}:latest-${GO_VERSION}
```
Comment (jakexks) [inline images/golang-aws/build.yaml:30]:
```suggestion
- ${_REGISTRY}/${_NAME}:latest
```

Latest is special cased so docker pull without any tag will still work

**Review by jakexks — APPROVED** [review]

**Pull request merged**